cub/test/catch2_test_device_run_length_encode_non_trivial_runs.cu

```diff
@@ -368,3 +388,98 @@ C2H_TEST("DeviceRunLengthEncode::NonTrivialRuns does not run out of memory", "[d
+  auto check_run_length_out_helper = detail::large_problem_test_helper(num_uniques);
+  auto expected_run_lengths_it     = cuda::make_constant_iterator(run_length_type{2});
+  auto check_run_length_out_it     = check_run_length_out_helper.get_flagging_output_iterator(expected_run_lengths_it);
+
+  // Allocate memory for the number of expected unique items
+  c2h::device_vector<offset_type> out_num_runs(1);
+
+  // Run algorithm under test
+  run_length_encode(input_item_it, check_offset_out_it, check_run_length_out_it, out_num_runs.begin(), num_items);
+
+  // Verify result
+  REQUIRE(out_num_runs[0] == num_uniques);
+  check_offset_out_helper.check_all_results_correct();
+  check_run_length_out_helper.check_all_results_correct();
+}
+catch (std::bad_alloc& e)
+{
+  std::cerr << "Caught bad_alloc: " << e.what() << std::endl;
+}
+
+C2H_TEST("DeviceRunLengthEncode::NonTrivialRuns works for large runs of equal items",
+         "[device][run_length_encode][skip-cs-initcheck][skip-cs-racecheck][skip-cs-synccheck][!mayfail]",
+         offset_types)
+try
+{
+  using offset_type     = typename c2h::get<0, TestType>;
+  using run_length_type = offset_type;
+  using item_t          = char;
+
+  CAPTURE(c2h::type_name<offset_type>(), c2h::type_name<run_length_type>(), c2h::type_name<item_t>());
+
+  const auto num_items = detail::make_large_offset<offset_type>();
+
+  constexpr auto num_uniques               = offset_type{2};
+  constexpr run_length_type first_run_size = 200;
+  const run_length_type second_run_size    = static_cast<run_length_type>(num_items) - first_run_size;
+
+  // First run is a small run of equal items
+  auto small_segment_it = cuda::make_constant_iterator(item_t{3});
+  // Second run is a very large run of equal items
+  auto large_segment_it = cuda::make_constant_iterator(item_t{42});
+  auto input_item_it    = detail::make_concat_iterators_op(small_segment_it, large_segment_it, first_run_size);
+
+  // Allocate some memory for the results
+  c2h::device_vector<offset_type> offsets_out(num_uniques, offset_type{42});
+  c2h::device_vector<run_length_type> run_lengths_out(num_uniques, run_length_type{42});
+
+  // Allocate memory for the number of expected unique items
+  c2h::device_vector<offset_type> out_num_runs(1);
+
+  // Run algorithm under test
+  // TODO (elstehle): Remove static_cast<int> in PR that adds support for large number of items
+  run_length_encode(
+    input_item_it, offsets_out.begin(), run_lengths_out.begin(), out_num_runs.begin(), static_cast<int>(num_items));
+
+  // Expected results
+  c2h::device_vector<offset_type> expected_uniques{offset_type{0}, first_run_size};
+  c2h::device_vector<run_length_type> expected_run_lengths{first_run_size, second_run_size};
+
+  // Verify result
+  REQUIRE(out_num_runs[0] == num_uniques);
+  REQUIRE(expected_uniques == offsets_out);
+  REQUIRE(expected_run_lengths == run_lengths_out);
```

Comment:
Suggestion: In case of a failure, it may help to show the results of all output variables:
```suggestion
  CHECK(out_num_runs[0] == num_uniques);
  CHECK(expected_uniques == offsets_out);
  CHECK(expected_run_lengths == run_lengths_out);
```